bin/faucet/src/client.rs

```diff
@@ -175,23 +178,24 @@ pub struct FaucetDataStore {
 
 impl FaucetDataStore {
     pub fn new(
-        faucet_account: Rc<RefCell<Account>>,
+        faucet_account: Account,
         seed: Word,
         root_block_header: BlockHeader,
         root_chain_mmr: ChainMmr,
     ) -> Self {
         Self {
-            faucet_account,
+            faucet_account: Mutex::new(faucet_account),
             seed,
             block_header: root_block_header,
             chain_mmr: root_chain_mmr,
         }
     }
 
     /// Updates the stored faucet account with the provided delta.
-    fn update_faucet_account(&mut self, delta: &AccountDelta) -> Result<(), ProcessError> {
+    fn update_faucet_account(&self, delta: &AccountDelta) -> Result<(), ProcessError> {
         self.faucet_account
-            .borrow_mut()
+            .lock()
+            .unwrap()
```

Comment:
I think we should avoid using `unwrap` in favor of `expect` everywhere except tests if panic is highly unlikely to happen (otherwise, an error should be returned). This helps us to understand why method might fail or why we don't expect it to fail. I usually add `.expect("Poisoned lock")` for every `lock` or `read`/`write` in case of using `RwLock`.